docs/guides/directing-traffic-to-your-app.md

```diff
@@ -0,0 +1,375 @@
+hostname `cafe.example.com` from the Gateway to the coffee Service.
+
+Once NGINX Kubernetes Gateway parses the `cafe` Gateway and `coffee` HTTPRoute, it will configure its dataplane, NGINX,
+to route all HTTP requests to `cafe.example.com` to the Pods that the `coffee` Service targets:
+
+![Traffic Flow](/docs/images/direct-all-traffic-flow.png)
+
+The coffee Service is omitted from the diagram above because the NGINX Kubernetes Gateway routes directly to the Pods
+that the coffee Service targets.
+
+> **Note**
+> In the diagrams above, all resources that are the responsibility of the cluster operator are shown in green.
+> The orange resources are the responsibility of the application developers.
+> See the [roles and personas](https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/#roles-and-personas_1)
+> Gateway API document for more information on these roles.
+
+## Create the Gateway API Resources
+
+To create the `cafe` Gateway, copy and paste the following into your terminal:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: cafe
+spec:
+  gatewayClassName: nginx
+  listeners:
+  - name: http
+    port: 80
+    protocol: HTTP
+EOF
+```
+
+This Gateway is associated with the NGINX Kubernetes Gateway through the `gatewayClassName` field. The default
+installation of NGINX Kubernetes Gateway creates a GatewayClass with the name `nginx`. NGINX Kubernetes Gateway will
+only configure Gateways with a `gatewayClassName` of `nginx` unless you change the name via the `--gatewayclass`
+[command-line flag](/docs/cli-help.md#static-mode).
+
+We specify a listener on the Gateway to open an access point on the cluster. In this case, since the coffee application
+accepts HTTP requests, we create an HTTP listener, named `http`, that listens on port 80.
+
+By default, Gateways only allow routes (such as HTTPRoutes) to attach if they are in the same namespace as the Gateway.
+If you want to change this behavior, you can set
+the [`allowedRoutes`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.AllowedRoutes)
+field.
+
+Now, let's create the HTTPRoute by copying and pasting the following into your terminal:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: coffee
+spec:
+  parentRefs:
+  - name: cafe
+  hostnames:
+  - "cafe.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /
+    backendRefs:
+    - name: coffee
+      port: 80
+EOF
+```
+
+To attach the `coffee` HTTPRoute to the `cafe` Gateway, we specify the Gateway name in the `parentRefs` field. The
+attachment will succeed if the hostnames and protocol in the HTTPRoute are allowed by at least one of the
+Gateway's listeners.
+
+The `hostnames` field allows you to list the hostnames that the HTTPRoute matches. In this case, incoming requests
```

Comment:
Not sure. I'd like to minimize the number of links throughout the doc so that the user can read all the way through without opening extra tabs/windows. 

Is there context that you think we are missing around hostnames and rules?

cc @mpstefan

Reply:
I was thinking from the point of view they have an option to view a link, not  that they shall visit it - we don't say, "see this link" explicitly.

but if it is perceived that way, better not overload the user

Reply:
I second the viewpoint that we should give them the option to visit by hyperlinking words. If they already know the context, there's no reason for them to click. If they are confused, they can click on it to get more context.

I see this a lot in the [Kubernetes documentation](https://kubernetes.io/docs/concepts/services-networking/service/) and I really appreciate the references. Otherwise, I'd have to do another Google search to get the full picture for some concepts.